platform_api/orchestrator/kube_client.py

```diff
@@ -1575,25 +1590,27 @@ def are_sufficient(self, pod: PodDescriptor) -> bool:
         return (
             self.cpu_mcores >= r.cpu_mcores
             and self.memory >= r.memory
-            and self.gpu >= (r.gpu or 0)
+            and self.nvidia_gpu >= (r.nvidia_gpu or 0)
         )
 
     def __add__(self, other: "NodeResources") -> "NodeResources":
         return self.__class__(
             cpu=self.cpu + other.cpu,
             memory=self.memory + other.memory,
-            gpu=self.gpu + other.gpu,
+            nvidia_gpu=self.nvidia_gpu + other.nvidia_gpu,
+            amd_gpu=self.amd_gpu + other.amd_gpu,
         )
 
     def __sub__(self, other: "NodeResources") -> "NodeResources":
         return self.__class__(
```

Comment:
nit: maybe we could wrap this method calls with `try:expect` and return empty resources for cases, when node is overcommited? previously it led to poller stuck in crash loop of trying to calculate `get_free_resources` for the node

we could do that separately, or I could do it so we do not mix PRs

Reply:
let's solve it in a separate pr. there is issue for this https://linear.app/apolo/issue/ENG-35/platform-api-poller-should-not-crash-when-there-are-evicted-pods-on